changing class name to BEM standards

NataliaLKB opened the pull request:
@johnduffell regarding changes talked about in https://github.com/guardian/frontend/pull/10319.

Just changing the classname, and ensuring that the svg is smaller when needed.

davidrapson commented:
In this case we don't have a base `.inline-close` class so the `.inline-close--small` modifier feels slightly odd. What is the purpose of this? It looks like it's a modifier class for an `.inline-icon`, if so `.inline-icon--close` might make most sense. If not a standalone class of `.inline-close` might be clearer.

The point is we're currently modifying something that doesn't have a base class, which probably suggests there's a clearer name.

Sorry for bike-shedding :bike: on something so small. :+1: Definitely preferable to the chaining.

uplne commented:
:+1:

uplne commented:
@davidrapson if I remember correctly the inlinesvg method will add class .inline- + whatever is passed as parameter. In this case it will be .inline-close. Plus extra classes from List. So it should be ok.

NataliaLKB commented:
@davidrapson Please ask and comment away! It is great, especially because I am not accustomed to BEM syntax yet.

`.inline-icon` is a generic classname added for almost all inline svgs when added like [this](https://github.com/guardian/frontend/blob/bem-syntax/common/app/views/fragments/message.scala.html#L15). `.inline-close` is also added in this case because of the name of the icon. 

The reason why I need to have a small version of this icon is because previously we had two svgs that were essentially doing the same job and I have simplified it by us only using one and taking advantage of the scalability of svgs.

NataliaLKB commented:
@uplne exactly! Thanks :)

davidrapson commented:
Ah I see, so this `inline-@file` part in https://github.com/guardian/frontend/blob/bem-syntax/common/app/views/fragments/inlineSvg.scala.html#L2 right?

In that case your `.inline-close--small` class makes perfect sense. Generated classes add a bit of abstraction which is why I missed it. Perhaps add a comment explaining that the `.inline-close` icon is inferred. Don't let me stop you shipping though :+1: :ship:

NataliaLKB commented:
@davidrapson Yes it may help to add more comments around this. I will be making a new PR soon with more of these sorts of edits so I will have a think on how to make it more clear for them. 

Thanks again!
